parser/parse_frontmatter_test.go

```diff
@@ -200,6 +200,10 @@ func TestPageHasFrontMatter(t *testing.T) {
 		{[]byte("---"), false},
 		{[]byte("---\n"), true},
 		{[]byte("---\n"), true},
+		{[]byte("--- \n"), true},
+		{[]byte("---  \n"), true},
+		{[]byte("--- \r\n"), true},
+		{[]byte("---  \r\n"), true},
```

Comment:
I think it doesn't need to add `\r\n` case because at

```go
test.content = bytes.Replace(test.content, []byte("\n"), []byte(ending), -1)
```

line, `\n` characters in all test cases are replaced by `\n` or `\r\n` and both cases are tested.